funsor/contract.py

```diff
@@ -45,11 +41,17 @@ def _simplify_contract(lhs, rhs, reduced_vars):
         lhs = lhs.reduce(ops.add, reduced_vars - rhs_vars)
         reduced_vars = reduced_vars & rhs_vars
         progress = True
-
     if progress:
         return Contract(lhs, rhs, reduced_vars)
 
-    return None
+    return fn(lhs, rhs, reduced_vars)
+
+
+def contractor(fn):
```

Comment:
I've made this a little easier to use, since we're now importing it in other files.